app/controllers/foreman_snapshot_management/snapshots_controller.rb

```diff
@@ -67,8 +72,36 @@ def update
       end
     end
 
+    define_action_permission ['select_multiple_host', 'create_multiple_host'], :create
+    def select_multiple_host; end
+
+    def create_multiple_host
+      data = snapshot_params
+      snapshots_created = 0
+      @hosts.each do |h|
+        s = Snapshot.new(data.merge(host: h))
+        if s.create
+          snapshots_created += 1
+        else
+          error _('Error occurred while creating Snapshot for "%{name}": %{msg}') % { name: h.name, msg: s.errors.full_messages.to_sentence }
+        end
+      end
+      if snapshots_created > 0
+        success _('Created %{snapshots} for %{num} %{hosts}') % {
+          snapshots: n_('Snapshot', 'Snapshots', snapshots_created),
+          num: snapshots_created,
+          hosts: n_('host', 'hosts', snapshots_created)
+        }
```

Comment:
How about adding an explicit return here?

Reply:
I am not sure what you mean.
Is the return not covered by the `redirect_back_or_to`?

Reply:
I think it's fine in the current version ;-)